### PR TITLE
platforms:tgl: Configure format converter for IPC4 when building for Zephyr

### PR DESCRIPTION
**Pull request by XiaoyunWu6666**
Enable format converters when building tgl firmware with Zephyr and IPC4.

Signed-off-by: Iris Wu <xiaoyun.wu@intel.com>

### CODE REVIEW

File: platforms/tgl/ipc4_overlay.conf [hunk 1]
```diff
@@ -1 +1,14 @@
 CONFIG_IPC_MAJOR_4=y
+CONFIG_RIMAGE_SIGNING_SCHEMA="tgl-cavs"
+CONFIG_FORMAT_CONVERT_HIFI3=n
+CONFIG_PCM_CONVERTER_FORMAT_CONVERT_HIFI3=n
+CONFIG_FORMAT_S16LE=y
```
Comment (RanderWang) [inline platforms/tgl/ipc4_overlay.conf:5]: S16LE is enabled by default so remove it

Comment (RanderWang) [reply platforms/tgl/ipc4_overlay.conf:5]: add "CONFIG_PCM_CONVERTER_FORMAT_S24_3LE=y", thanks

**Review by lgirdwood — COMMENTED** [review]
Why do we disable HiFi3 ?

**Comment by RanderWang** [comment]
> Why do we disable HiFi3 ?

Currently ipc4 format conversion doesn't support HIFI3, so only generic format conversion is supported. The issue has been temporarily addressed by bf191f6b67a0b9bc621e34d87504fa6a8f34b68e
@XiaoyunWu6666  can you remove HIFI=n and test ?

**Comment by lgirdwood** [comment]
@singalsu how much effort to add the HiFi3 PCM converter support ? This will give some parity when testing IPC4 code and topologies on TGL/ADL.

**Review by lgirdwood — APPROVED** [review]

**Comment by singalsu** [comment]
> @singalsu how much effort to add the HiFi3 PCM converter support ? This will give some parity when testing IPC4 code and topologies on TGL/ADL.

@RanderWang @lgirdwood Can you provide more details about what is the missing piece in IPC4 HiFi3 pcm conversion. Also I find it hard to test anything IPC4 related, no testbench, no cmocka. E.g. the IPC4 only components.

**Comment by lgirdwood** [comment]
> > @singalsu how much effort to add the HiFi3 PCM converter support ? This will give some parity when testing IPC4 code and topologies on TGL/ADL.
> 
> @RanderWang @lgirdwood Can you provide more details about what is the missing piece in IPC4 HiFi3 pcm conversion. Also I find it hard to test anything IPC4 related, no testbench, no cmocka. E.g. the IPC4 only components.

@singalsu IIUC the PCM converter in IPC4 uses HiFi4 and converts all the PCM formats for Windows, but the Hifi3 version is missing some of the format conversions. Please take a quick check of the Hifi4 code and see if the missing conversions are obvious. Thanks

**Review by RanderWang — APPROVED** [review]

**Pull request merged**